src/rust/ensogl/src/display/scene.rs

```diff
@@ -551,7 +553,14 @@ pub struct Renderer {
 }
 
 impl Renderer {
-    pub fn new(logger:&Logger, dom:&Dom, context:&Context, variables:&UniformScope) -> Self {
+
+    pub fn new
+    (logger:impl Into<RendererLogger>, dom:&Dom, context:&Context, variables:&UniformScope) -> Self {
+        Self::new_impl(logger.into(),dom,context,variables)
+    }
+
+    /// Separate implementation of `new` to avoid function body duplication after logger monorphization.
+    fn new_impl(logger:RendererLogger, dom:&Dom, context:&Context, variables:&UniformScope) -> Self {
```

Comment:
By default We should not worry about code size of generics.

Unless this split actaully does make any difference. I doubt, but check the actual wasm size after minimization and if there is visible difference, explain this in comment.